Fix ESLint issues

#5455 enables ESLint for the codebase, but some of the tests have been disabled for now because ESLint seems to be better at finding issues than JSHint. This is the list of currently disabled rules that either need to be fixed, or should be removed from the config:

- [ ] [`no-proto`](http://eslint.org/docs/rules/no-proto.html) (JSHint `proto`)
- [x] [`strict`](http://eslint.org/docs/rules/strict.html) (JSHint `strict`)
- [x] [`indent`](http://eslint.org/docs/rules/indent.html) (JSHint `indent`)
- [ ] [`quotes`](http://eslint.org/docs/rules/quotes.html) (JSHint `quotmark`)
- [ ] [`dot-notation`](http://eslint.org/docs/rules/dot-notation.html) (JSHint `sub`)

```
/Users/tbn/Code/ember-cli/lib/broccoli/ember-addon.js
  57:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/ext/promise.js
  19:3  warning  The '__proto__' property is deprecated  no-proto
  38:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/addon-discovery.js
  28:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/addon.js
  116:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/addons-factory.js
  23:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/blueprint.js
  288:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/command.js
  92:1  warning  The '__proto__' property is deprecated  no-proto

/Users/tbn/Code/ember-cli/lib/models/task.js
  11:1  warning  The '__proto__' property is deprecated  no-proto
```

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -85,9 +85,6 @@ module.exports = {
     // JSHint "unused"
     'no-unused-vars': 2,
 
-    // JSHint "sub", disabled due to warnings
-    'dot-notation': 0,
-
     // JSHint "trailing"
     'no-trailing-spaces': 2,
 

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -23,7 +23,7 @@ module.exports = {
     // JSHint "strict"
     'strict': [2, 'global'],
 
-    // JSHint "indent", disabled due to warnings
+    // JSHint "indent"
     'indent': [2, 2, {
       'SwitchCase': 1,
       'VariableDeclarator': { 'var': 2, 'let': 2, 'const': 3 }

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -17,9 +17,6 @@ module.exports = {
       allowTernary: true,
     }],
 
-    // JSHint "proto", disabled due to warnings
-    'no-proto': 0,
-
     // JSHint "strict"
     'strict': [2, 'global'],
 

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -70,8 +70,8 @@ module.exports = {
     // JSHint "noempty", JSCS "disallowEmptyBlocks"
     'no-empty': 2,
 
-    // JSHint "quotmark", disabled due to warnings
-    quotes: [0, 'single'],
+    // JSHint "quotmark"
+    quotes: [2, 'single'],
 
     // JSHint "nonew", disabled already in .jshintrc
     'no-new': 0,

--- a/blueprints/acceptance-test/index.js
+++ b/blueprints/acceptance-test/index.js
@@ -14,7 +14,7 @@ module.exports = {
     }
     return {
       testFolderRoot: testFolderRoot,
-      friendlyTestName: testInfo.name(options.entity.name, "Acceptance", null)
+      friendlyTestName: testInfo.name(options.entity.name, 'Acceptance', null)
     };
   }
 };

--- a/blueprints/adapter-test/index.js
+++ b/blueprints/adapter-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates an ember-data adapter unit test',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Adapter")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Adapter')
     };
   }
 };

--- a/blueprints/component-test/index.js
+++ b/blueprints/component-test/index.js
@@ -39,15 +39,15 @@ module.exports = {
   locals: function(options) {
     var dasherizedModuleName = stringUtil.dasherize(options.entity.name);
     var componentPathName = dasherizedModuleName;
-    var testType = options.testType || "integration";
-    var friendlyTestDescription = testInfo.description(options.entity.name, "Integration", "Component");
+    var testType = options.testType || 'integration';
+    var friendlyTestDescription = testInfo.description(options.entity.name, 'Integration', 'Component');
 
     if (options.pod && options.path !== 'components' && options.path !== '') {
       componentPathName = [options.path, dasherizedModuleName].join('/');
     }
 
     if (options.testType === 'unit') {
-      friendlyTestDescription = testInfo.description(options.entity.name, "Unit", "Component");
+      friendlyTestDescription = testInfo.description(options.entity.name, 'Unit', 'Component');
     }
 
     return {

--- a/blueprints/controller-test/index.js
+++ b/blueprints/controller-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a controller unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Controller")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Controller')
     };
   }
 };

--- a/blueprints/helper-test/index.js
+++ b/blueprints/helper-test/index.js
@@ -7,7 +7,7 @@ module.exports = {
   description: 'Generates a helper unit test.',
   locals: function(options) {
     return {
-      friendlyTestName: testInfo.name(options.entity.name, "Unit", "Helper"),
+      friendlyTestName: testInfo.name(options.entity.name, 'Unit', 'Helper'),
       dasherizedModulePrefix: stringUtils.dasherize(options.project.config().modulePrefix)
     };
   }

--- a/blueprints/initializer-test/index.js
+++ b/blueprints/initializer-test/index.js
@@ -7,7 +7,7 @@ module.exports = {
   description: 'Generates an initializer unit test.',
   locals: function(options) {
     return {
-      friendlyTestName: testInfo.name(options.entity.name, "Unit", "Initializer"),
+      friendlyTestName: testInfo.name(options.entity.name, 'Unit', 'Initializer'),
       dasherizedModulePrefix: stringUtils.dasherize(options.project.config().modulePrefix)
     };
   }

--- a/blueprints/instance-initializer-test/index.js
+++ b/blueprints/instance-initializer-test/index.js
@@ -7,7 +7,7 @@ module.exports = {
   description: 'Generates an instance initializer unit test.',
   locals: function(options) {
     return {
-      friendlyTestName: testInfo.name(options.entity.name, "Unit", "Instance Initializer"),
+      friendlyTestName: testInfo.name(options.entity.name, 'Unit', 'Instance Initializer'),
       dasherizedModulePrefix: stringUtils.dasherize(options.project.config().modulePrefix)
     };
   }

--- a/blueprints/model-test/index.js
+++ b/blueprints/model-test/index.js
@@ -9,7 +9,7 @@ module.exports = {
   locals: function(options) {
     var result = ModelBlueprint.locals.apply(this, arguments);
 
-    result.friendlyDescription = testInfo.description(options.entity.name, "Unit", "Model");
+    result.friendlyDescription = testInfo.description(options.entity.name, 'Unit', 'Model');
 
     return result;
   }

--- a/blueprints/model/index.js
+++ b/blueprints/model/index.js
@@ -44,7 +44,7 @@ module.exports = {
       }
 
       if (/has-many|belongs-to/.test(dasherizedType)) {
-        needs.push("'model:" + dasherizedForeignModelSingular + "'");
+        needs.push('\'model:' + dasherizedForeignModelSingular + '\'');
       }
     }
     var needsDeduplicated = needs.filter(function(need, i) {

--- a/blueprints/route-test/index.js
+++ b/blueprints/route-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a route unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Route")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Route')
     };
   },
 };

--- a/blueprints/serializer-test/index.js
+++ b/blueprints/serializer-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a serializer unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Serializer")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Serializer')
     };
   },
 };

--- a/blueprints/service-test/index.js
+++ b/blueprints/service-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a service unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Service")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Service')
     };
   },
 };

--- a/blueprints/transform-test/index.js
+++ b/blueprints/transform-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a transform unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "Transform")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'Transform')
     };
   },
 };

--- a/blueprints/util-test/index.js
+++ b/blueprints/util-test/index.js
@@ -7,7 +7,7 @@ module.exports = {
   description: 'Generates a util unit test.',
   locals: function(options) {
     return {
-      friendlyTestName: testInfo.name(options.entity.name, "Unit", "Utility"),
+      friendlyTestName: testInfo.name(options.entity.name, 'Unit', 'Utility'),
       dasherizedModulePrefix: stringUtils.dasherize(options.project.config().modulePrefix)
     };
   }

--- a/blueprints/view-test/index.js
+++ b/blueprints/view-test/index.js
@@ -6,7 +6,7 @@ module.exports = {
   description: 'Generates a view unit test.',
   locals: function(options) {
     return {
-      friendlyTestDescription: testInfo.description(options.entity.name, "Unit", "View")
+      friendlyTestDescription: testInfo.description(options.entity.name, 'Unit', 'View')
     };
   },
 };

--- a/tests/.eslintrc.js
+++ b/tests/.eslintrc.js
@@ -8,5 +8,8 @@ module.exports = {
 
     // JSHint "unused"
     'no-unused-vars': 0,
+
+    // disabled for easier asserting of file contents
+    'quotes': 0,
   }
 };

